libsolc/libsolc.cpp

```diff
@@ -42,11 +42,12 @@ ReadCallback::Callback wrapReadCallback(CStyleReadFileCallback _readCallback = n
 	ReadCallback::Callback readCallback;
 	if (_readCallback)
 	{
-		readCallback = [=](string const& _path)
+		readCallback = [=](string const& _path, ReadCallback::Kind _callbackKind = ReadCallback::Kind::ReadFile)
 		{
 			char* contents_c = nullptr;
 			char* error_c = nullptr;
-			_readCallback(_path.c_str(), &contents_c, &error_c);
+			string inputWithKind{ReadCallback::kindString(_callbackKind) + _path};
```

Comment:
This is still a breaking change. Anyone not using the wrapper will have to adapt.

It would not be a breaking change if the prefix would only be added in case the feature is requested through standard-json input or something like that.